packages/cli/api-importers/v2-importer-commons/src/converters/schema/OneOfSchemaConverter.ts

```diff
@@ -210,4 +191,105 @@ export class OneOfSchemaConverter extends AbstractConverter<
             inlinedTypes
         };
     }
+
+    private shouldConvertAsNullableSchemaOrReference(): boolean {
+        if (this.schema.oneOf != null) {
+            return this.schema.oneOf.some((subSchema) => "type" in subSchema && subSchema.type === "null");
+        } else if (this.schema.anyOf != null) {
+            return this.schema.anyOf.some((subSchema) => "type" in subSchema && subSchema.type === "null");
+        }
+        return false;
+    }
+
+    private simplifySchema({
+        context,
+        errorCollector
+    }: {
+        context: AbstractConverterContext<object>;
+        errorCollector: ErrorCollector;
+    }): OpenAPIV3_1.SchemaObject | undefined {
+        if (this.schema.oneOf != null) {
+            const oneOfWithoutNull = this.schema.oneOf.filter(
+                (subSchema) => "type" in subSchema && subSchema.type !== "null"
+            );
+            if (oneOfWithoutNull.length === 0) {
+                errorCollector.collect({
+                    message: `Received oneOf schema with no valid non-null types: ${JSON.stringify(this.schema)}`,
+                    path: this.breadcrumbs
+                });
+                return undefined;
+            }
+            if (oneOfWithoutNull.length === 1) {
+                return {
+                    ...this.schema,
+                    oneOf: undefined,
+                    ...oneOfWithoutNull[0]
+                };
+            } else {
+                return {
+                    ...this.schema,
+                    oneOf: oneOfWithoutNull
+                };
+            }
+        } else if (this.schema.anyOf != null) {
+            const anyOfWithoutNull = this.schema.anyOf.filter(
+                (subSchema) => "type" in subSchema && subSchema.type !== "null"
+            );
+            if (anyOfWithoutNull.length === 0) {
+                errorCollector.collect({
+                    message: `Received anyOf schema with no valid non-null types: ${JSON.stringify(this.schema)}`,
+                    path: this.breadcrumbs
+                });
+                return undefined;
+            }
+            if (anyOfWithoutNull.length === 1) {
+                return {
+                    ...this.schema,
+                    anyOf: undefined,
+                    ...anyOfWithoutNull[0]
+                };
+            } else {
+                return {
+                    ...this.schema,
+                    anyOf: anyOfWithoutNull
+                };
+            }
+        }
+        return undefined;
+    }
+
+    private async convertAsNullableSchemaOrReference({
+        context,
+        errorCollector
+    }: {
+        context: AbstractConverterContext<object>;
+        errorCollector: ErrorCollector;
+    }): Promise<OneOfSchemaConverter.Output | undefined> {
+        const simplifiedSchema = this.simplifySchema({ context, errorCollector });
```

Comment:
nit: change helper function to be more descriptive -- `this.removeNullFromAnyOfAndCoerceIntoSingleSchema`